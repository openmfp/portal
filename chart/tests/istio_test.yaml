suite: istio
templates:
  - istio-*
tests:
  - it: matches the snapshot
    values:
      - ../test-values.yaml
    asserts:
      - matchSnapshot: {}
  - it: matches the snapshot with istio disabled
    values:
      - ../test-values.yaml
    set:
      istio:
        enabled: false
    asserts:
      - matchSnapshot: {}
