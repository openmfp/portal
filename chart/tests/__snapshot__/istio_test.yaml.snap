matches the snapshot:
  1: |
    apiVersion: core.openmfp.io/v1alpha1
    kind: ContentConfiguration
    metadata:
      name: gardener
      namespace: NAMESPACE
    spec:
      inlineConfiguration:
        content: |2-
          {
            "name": "gardener",
            "creationTimestamp": "2022-05-17T11:37:17Z",
            "luigiConfigFragment": [
              {
                "data": {
                  "nodes": [
                    {
                      "entityType": "example",
                      "pathSegment": "gardener",
                      "label": "Gardener Dashboard",
                      "virtualTree": true,
                      "initialRoute": "namespace/_all/shoots",
                      "url": "https://d.ing.gardener-op.mfp-dev.shoot.canary.k8s-hana.ondemand.com",
                      "icon": "https://d.ing.gardener-op.mfp-dev.shoot.canary.k8s-hana.ondemand.com/static/assets/logo.svg"
                    }
                  ]
                }
              }
            ]
          }
        contentType: json
  2: |
    apiVersion: core.openmfp.io/v1alpha1
    kind: ContentConfiguration
    metadata:
      name: home
      namespace: NAMESPACE
    spec:
      inlineConfiguration:
        content: "\n{\n  \"name\": \"overview\",\n  \"luigiConfigFragment\": [\n    {\n      \"data\": {\n        \"nodes\": [\n          {\n            \"entityType\": \"global\",\n            \"pathSegment\": \"home\",\n            \"label\": \"Overview\",\n            \"icon\": \"home\",\n            \"hideFromNav\": true,            \n            \"defineEntity\": {\n              \"id\": \"example\"\n            },\n            \"children\": [\n              {\n                \"pathSegment\": \"overview\",\n                \"label\": \"Overview\",\n                \"icon\": \"home\",\n                \"url\": \"https://fiddle.luigi-project.io/examples/microfrontends/multipurpose.html\",\n                \"context\": {\n                  \"title\": \"Welcome to OpenMFP Portal\",\n                  \"content\": \" \"\n                }\n              }\n            ]\n          }\n        ]\n      }\n    }\n  ]\n}"
        contentType: json
  3: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: RELEASE-NAME
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: RELEASE-NAME
      strategy:
        rollingUpdate:
          maxSurge: 5
          maxUnavailable: 0
      template:
        metadata:
          labels:
            app: RELEASE-NAME
        spec:
          containers:
            - env:
                - name: OIDC_CLIENT_ID_PORTAL
                  value: 12345
                - name: OIDC_CLIENT_SECRET_PORTAL
                  valueFrom:
                    secretKeyRef:
                      key: secret
                      name: portal-client-secret-portal
                - name: AUTH_SERVER_URL_PORTAL
                  value: https://auth.example.com
                - name: BASE_DOMAINS_PORTAL
                  value: portal.example.com,portal2.example.com
                - name: IDP_NAMES
                  value: portal
                - name: PORT
                  value: "8080"
                - name: REGION
                  value: null
                - name: IMAGE_TAG
                  value: "12345"
                - name: IMAGE_NAME
                  value: ghcr.io/openmfp/portal
                - name: COOKIE_DOMAIN
                  value: null
              image: ghcr.io/openmfp/portal:12345
              imagePullPolicy: IfNotPresent
              name: RELEASE-NAME
              ports:
                - containerPort: 8080
                  name: http
              resources:
                limits:
                  cpu: "1"
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
          securityContext:
            fsGroup: 2000
            runAsGroup: 3000
            runAsUser: 1000
          serviceAccountName: RELEASE-NAME
  4: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      name: RELEASE-NAME-portal-client-secret-portal
      namespace: NAMESPACE
    spec:
      data:
        - remoteRef:
            conversionStrategy: Default
            key: dxp-core-team/manual-secrets/portal-client-secrets
            property: portal
          secretKey: secret
      refreshInterval: 10m
      secretStoreRef:
        kind: SecretStore
        name: environment-store
      target:
        creationPolicy: Owner
        deletionPolicy: Retain
        name: portal-client-secret-portal
  5: |
    apiVersion: networking.istio.io/v1alpha3
    kind: DestinationRule
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      host: RELEASE-NAME.NAMESPACE.svc.cluster.local
      trafficPolicy:
        tls:
          mode: ISTIO_MUTUAL
  6: |
    apiVersion: security.istio.io/v1beta1
    kind: PeerAuthentication
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      mtls:
        mode: STRICT
      selector:
        matchLabels:
          app: RELEASE-NAME
  7: |
    apiVersion: v1
    imagePullSecrets:
      - name: github
    kind: ServiceAccount
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
  8: |
    apiVersion: v1
    kind: Service
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      ports:
        - name: http
          port: 8080
          protocol: TCP
      selector:
        app: RELEASE-NAME
      type: ClusterIP
  9: |
    apiVersion: networking.istio.io/v1beta1
    kind: Sidecar
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      outboundTrafficPolicy:
        mode: ALLOW_ANY
      workloadSelector:
        labels:
          app: RELEASE-NAME
  10: |
    apiVersion: networking.istio.io/v1alpha3
    kind: VirtualService
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      gateways:
        - NAMESPACE/gateway
      hosts:
        - portal.example.com
        - portal.portal.example.com
        - portal2.example.com
        - portal.portal2.example.com
      http:
        - corsPolicy:
            allowHeaders:
              - Authorization
              - Content-Type
              - '*'
            allowMethods:
              - GET
              - POST
              - PUT
              - DELETE
          route:
            - destination:
                host: RELEASE-NAME.NAMESPACE.svc.cluster.local
                port:
                  number: 8080
matches the snapshot with istio disabled:
  1: |
    apiVersion: core.openmfp.io/v1alpha1
    kind: ContentConfiguration
    metadata:
      name: gardener
      namespace: NAMESPACE
    spec:
      inlineConfiguration:
        content: |2-
          {
            "name": "gardener",
            "creationTimestamp": "2022-05-17T11:37:17Z",
            "luigiConfigFragment": [
              {
                "data": {
                  "nodes": [
                    {
                      "entityType": "example",
                      "pathSegment": "gardener",
                      "label": "Gardener Dashboard",
                      "virtualTree": true,
                      "initialRoute": "namespace/_all/shoots",
                      "url": "https://d.ing.gardener-op.mfp-dev.shoot.canary.k8s-hana.ondemand.com",
                      "icon": "https://d.ing.gardener-op.mfp-dev.shoot.canary.k8s-hana.ondemand.com/static/assets/logo.svg"
                    }
                  ]
                }
              }
            ]
          }
        contentType: json
  2: |
    apiVersion: core.openmfp.io/v1alpha1
    kind: ContentConfiguration
    metadata:
      name: home
      namespace: NAMESPACE
    spec:
      inlineConfiguration:
        content: "\n{\n  \"name\": \"overview\",\n  \"luigiConfigFragment\": [\n    {\n      \"data\": {\n        \"nodes\": [\n          {\n            \"entityType\": \"global\",\n            \"pathSegment\": \"home\",\n            \"label\": \"Overview\",\n            \"icon\": \"home\",\n            \"hideFromNav\": true,            \n            \"defineEntity\": {\n              \"id\": \"example\"\n            },\n            \"children\": [\n              {\n                \"pathSegment\": \"overview\",\n                \"label\": \"Overview\",\n                \"icon\": \"home\",\n                \"url\": \"https://fiddle.luigi-project.io/examples/microfrontends/multipurpose.html\",\n                \"context\": {\n                  \"title\": \"Welcome to OpenMFP Portal\",\n                  \"content\": \" \"\n                }\n              }\n            ]\n          }\n        ]\n      }\n    }\n  ]\n}"
        contentType: json
  3: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: RELEASE-NAME
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: RELEASE-NAME
      strategy:
        rollingUpdate:
          maxSurge: 5
          maxUnavailable: 0
      template:
        metadata:
          labels:
            app: RELEASE-NAME
        spec:
          containers:
            - env:
                - name: OIDC_CLIENT_ID_PORTAL
                  value: 12345
                - name: OIDC_CLIENT_SECRET_PORTAL
                  valueFrom:
                    secretKeyRef:
                      key: secret
                      name: portal-client-secret-portal
                - name: AUTH_SERVER_URL_PORTAL
                  value: https://auth.example.com
                - name: BASE_DOMAINS_PORTAL
                  value: portal.example.com,portal2.example.com
                - name: IDP_NAMES
                  value: portal
                - name: PORT
                  value: "8080"
                - name: REGION
                  value: null
                - name: IMAGE_TAG
                  value: "12345"
                - name: IMAGE_NAME
                  value: ghcr.io/openmfp/portal
                - name: COOKIE_DOMAIN
                  value: null
              image: ghcr.io/openmfp/portal:12345
              imagePullPolicy: IfNotPresent
              name: RELEASE-NAME
              ports:
                - containerPort: 8080
                  name: http
              resources:
                limits:
                  cpu: "1"
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
          securityContext:
            fsGroup: 2000
            runAsGroup: 3000
            runAsUser: 1000
          serviceAccountName: RELEASE-NAME
  4: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      name: RELEASE-NAME-portal-client-secret-portal
      namespace: NAMESPACE
    spec:
      data:
        - remoteRef:
            conversionStrategy: Default
            key: dxp-core-team/manual-secrets/portal-client-secrets
            property: portal
          secretKey: secret
      refreshInterval: 10m
      secretStoreRef:
        kind: SecretStore
        name: environment-store
      target:
        creationPolicy: Owner
        deletionPolicy: Retain
        name: portal-client-secret-portal
  5: |
    apiVersion: v1
    imagePullSecrets:
      - name: github
    kind: ServiceAccount
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
  6: |
    apiVersion: v1
    kind: Service
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      ports:
        - name: http
          port: 8080
          protocol: TCP
      selector:
        app: RELEASE-NAME
      type: ClusterIP
